apps/extension/src/components/locked-panel/wallet-selector/index.tsx

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useState } from 'react'
+import { ChevronDownIcon, ChevronUpIcon } from '@radix-ui/react-icons'
+import { HardwareWalletIcon } from 'ui/src/components/icons'
+
+import { ToolTip } from 'ui/src/components/tool-tip'
+import { KeystoreType } from '@src/store/types'
+import {
+	Select,
+	SelectTrigger,
+	SelectValue,
+	SelectContent,
+	SelectViewport,
+	SelectItem,
+	SelectItemText,
+	SelectItemIndicator,
+	SelectScrollUpButton,
+	SelectScrollDownButton,
+} from 'ui/src/components/select'
+import { useSharedStore } from '@src/store'
+import { Box, Text, Flex } from 'ui/src/components/atoms'
+import Button from 'ui/src/components/button'
+import useMeasure from 'react-use-measure'
+
+interface IProps {}
+
+const defaultProps: Partial<IProps> = {
+	pool: null,
+}
+
+export const WalletSelector: React.FC<IProps> = () => {
+	const [open, setOpen] = useState<boolean>(false)
+	const [measureRef, { width: triggerWidth }] = useMeasure()
+
+	const { keystore, keystores, keystoreId, selectKeystore } = useSharedStore(state => ({
+		keystore: state.keystores.find(({ id }) => id === state.selectKeystoreId),
+		keystores: state.keystores,
+		keystoreId: state.selectKeystoreId,
+		selectKeystore: state.selectKeystoreAction,
+	}))
+
+	const isHardwareWallet = keystore.type === KeystoreType.HARDWARE
```

Comment:
in case keystore is not found

```suggestion
	const isHardwareWallet = keystore?.type === KeystoreType.HARDWARE
```